sdk/typescript/src/providers/provider.ts

```diff
@@ -362,4 +366,42 @@ export abstract class Provider {
    */
   abstract getReferenceGasPrice(): Promise<number>;
   // TODO: add more interface methods
+
+  /**
+   * Get the sequence number of the latest checkpoint that has been executed
+   */
+  abstract getLatestCheckpointSequenceNumber(): Promise<number>;
+
+  /**
+   * Returns checkpoint summary based on a checkpoint sequence number
+   * @param sequence_number - The sequence number of the desired checkpoint summary
+   */
+  abstract getCheckpointSummary(
+    sequence_number: number
```

Comment:
```suggestion
    sequenceNumber: number
```

nit: the convention is to use Camel case in TS